### PR TITLE
cli: add --system param to allow validating system user-data on a machine

### PR DESCRIPTION
**Pull request by blackboxsw**
## Test Steps
```bash
lxc launch ubuntu-daily:xenial dev-x
lxc file push cloudinit/config/schema.py dev-x/usr/lib/python3/dist-packages/cloudinit/config/
lxc exec dev-x bash
$ cloud-init devel schema --system
```

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly: **NOTE:** separating auto-generated documentation for all CLI commands to a separate PR

### CODE REVIEW

File: cloudinit/config/schema.py
Comment (mitechie) [inline cloudinit/config/schema.py:447]: shouldn't there be associated doc updates with adding a new argument to UI?

File: doc/rtd/topics/faq.rst
Comment (mitechie) [inline doc/rtd/topics/faq.rst:140]: This is the doc update but it seemed an edit to an existing doc and didn't seem clear to me.

**Review by OddBloke — CHANGES_REQUESTED** [review]
Thanks Chad, this is a nice little UX improvement. A couple of inline comments.

File: cloudinit/config/schema.py
Comment (blackboxsw) [reply cloudinit/config/schema.py:192]: @OddBloke The reason we treat this specially is because the typical userdata path in some cases is actually a compressed binary (juju deployed units). In this case the query command uses the special [cloudinit.cmd.query.load_userdata func to decompress and decode](https://github.com/canonical/cloud-init/blob/master/cloudinit/cmd/query.py#L81-L93) that I wanted to leverage here. It would probably be preferable to either call the function directly or relocate that utility function  into util.py.

File: doc/rtd/topics/faq.rst
Comment (blackboxsw) [reply doc/rtd/topics/faq.rst:140]: Yeah, devel commands, and general CLI documentation, isn't great. In the devel case I had originally left it fairly undocumented as these `devel` commands will likely change form or break compatibility before we decide to make them fully fledged top-level (non-devel) subcommands of cloud-init.  But, now that we are referencing subcommand switches in docs in other places. it would be helpful to have a more full rendering of cli docs for this devel subcommands. I ultimately think we should autogeerate docs from the argparse configs so that we can keep that up to date any time the cli is changed.  I think `sphinx-argparse`  extension could do this for us, but I'd like to leave that to a separate PR as it'll potentially involve re-organizing the top-level [CLI docs](https://cloudinit.readthedocs.io/en/latest/topics/cli.html) docs. As I think we could try to modify how the CLI docs in general are exposed and documented.

Comment (blackboxsw) [reply doc/rtd/topics/faq.rst:140]: That said, I did just push a commit that renders cmdline devel schema to this branch. I think we can build on this if an acceptable format for all CLI commands.

**Comment by blackboxsw** [comment]
@OddBloke @mitechie I've updated this branch and added auto-generated docs for review for only the devel subcommand using sphinx-argparse. If this documentation approach is palatable. I can add a followup PR to auto-gen docs for our entire CLI as a followup PR. (I could separate that from this PR too).

**Comment by blackboxsw** [comment]
@OddBloke @mitechie separated doc changes into a followup PR as discussed in standup yesterday

**Review by mitechie — APPROVED** [review]
With a +1 to Dan's comment on the file path at the start of the PR.

File: cloudinit/config/schema.py
Comment (blackboxsw) [reply cloudinit/config/schema.py:43]: @mitechie @OddBloke OOPS while the code no longer relies on this static var, I had forgotten to remove it.
Done.

File: cloudinit/config/schema.py
Comment (OddBloke) [inline cloudinit/config/schema.py:475]: Obviously a nit, but `print` takes `*args` and prints them:

```suggestion
            print("Valid cloud-config:", cfg_name)
```

```python
In [1]: print("a", "b")                                                                                                                                                                                                                                                                   
a b
```

File: cloudinit/config/schema.py
Comment (OddBloke) [inline cloudinit/config/schema.py:458]: I think this check needs updating: `all(exclusive_args)` will be `False` if any two of the three are passed:

```py
In [3]: exclusive_args = ["config_file", None, "system"]                                                                                                                                                                                                                                  

In [4]: if not any(exclusive_args) or all(exclusive_args): 
   ...:     print('Expected one of --config-file, --system or --docs arguments') 
   ...:      
In [5]:
```

**Review by OddBloke — CHANGES_REQUESTED** [review]
Thanks Chad! Overall this looks good, I have one inline comment that I think needs addressing before this can land (and another one which you can ignore because it's a nit :+1:).

File: cloudinit/config/schema.py
Comment (blackboxsw) [reply cloudinit/config/schema.py]: Good find. Changed logic and added unit tests

File: tests/unittests/test_handler/test_schema.py
Comment (OddBloke) [inline tests/unittests/test_handler/test_schema.py:424]: A nit (unless I'm wrong, then not even that): you could potentially fold this into the above definition by making the parameters `list(exclusive_combinations) + [()]`.

File: tests/unittests/test_handler/test_schema.py
Comment (OddBloke) [inline tests/unittests/test_handler/test_schema.py:409]: Nit: a not-particularly-important gap here is all three being specified at once.

File: cloudinit/config/schema.py
Comment (OddBloke) [inline cloudinit/config/schema.py:188]: This file is always root:root-owned by default and not world-readable, so this is not even a nit: we could instead rely on the user's permissions to determine if they can use this path.

**Review by OddBloke — APPROVED** [review]
Thanks Chad! Some nits inline, which you can fix in a follow-up if you so desire.

**Pull request merged**
